tests/specs/functional/storageRoot-spec.k

```diff
@@ -0,0 +1,32 @@
+requires "evm.k"
+requires "edsl.k"
+
+module VERIFICATION
+    imports EVM
+    imports EDSL
+
+    syntax StepSort ::= MerkleTree | String
+    syntax    KItem ::= runMerkle ( StepSort )
+                      | doneMerkle( StepSort )
+ // ------------------------------------------
+    rule runMerkle( T ) => doneMerkle( T )
+
+endmodule
+
+module STORAGEROOT-SPEC
+    imports VERIFICATION
+
+    rule <k> runMerkle( #storageRoot( .Map ) )
+          => doneMerkle( .MerkleTree )
+         </k>
+
+    rule <k> runMerkle( Keccak256( #rlpEncodeMerkleTree( #storageRoot(
+              #hashedLocation( "Solidity", keccakIntList(0), .IntList )                         |-> 1234
```

Comment:
According to [the ethereum wiki](https://github.com/ethereum/wiki/wiki/Patricia-Tree#storage-trie), every storage location is hashed. But it looks like in our implementation so far we have only been treating mapped locations as hashes while leaving regular positions as their raw value.

Reply:
See https://github.com/kframework/evm-semantics/blob/master/edsl.md#hashed-location-for-storage

I guess this depends on the implementation, so I may have to look more into what ganache is doing.

Reply:
@gtrepta it looks like (if you look at the definition of `#hashedLocation`) you can do `#hashedLocation("Solidity", 0, 0)` to get the functionality you want.

Reply:
It would actually be `#hashedLocation("Vyper", 0, 0)`. With "Solidity" it would do a `keccakIntList(0 0)`, which is wrong.